src/natcap/invest/validation.py

```diff
@@ -801,20 +882,16 @@ def validate(args, spec, spatial_overlap_opts=None):
         except KeyError:
             LOGGER.debug(f'Provided key {key} does not exist in ARGS_SPEC')
             continue
-        # If no validation options specified, assume defaults.
-        try:
-            validation_options = parameter_spec['validation_options']
-        except KeyError:
-            validation_options = {}
+
         type_validation_func = _VALIDATION_FUNCS[parameter_spec['type']]
+
         if type_validation_func is None:
             # Validation for 'other' type must be performed by the user.
             continue
-
         try:
-            warning_msg = type_validation_func(
-                args[key], **validation_options)
-
+            # pass the entire arg spec into the validation function as kwargs
+            # each type validation function allows extra kwargs with **kwargs
+            warning_msg = type_validation_func(args[key], **parameter_spec)
```

Comment:
Okay interesting. So the validation function extra **kwargs allows for validation options that we might not have implemented yet? Sorry I'm totally missing something here.

Reply:
Right, this allows us to pass in the entire spec for an arg, and ignore any spec properties that aren't used.